Result/Result.swift

```diff
@@ -131,6 +131,21 @@ public enum Result<T, Error: ErrorType>: CustomStringConvertible, CustomDebugStr
 	}
 }
 
+public protocol ResultType {
```

Comment:
Also, simple documentation comments for the protocol and its members would be greatly appreciated :bow: